aldryn_faq/tests/frontend/integration/specs/spec.faq.crud.js

```diff
@@ -35,4 +35,79 @@ describe('Aldryn FAQ tests: ', function () {
         });
     });
 
+    it('creates a new test page', function () {
+        // click the example.com link in the top menu
+        faqPage.userMenus.first().click().then(function () {
+            // wait for top menu dropdown options to appear
+            browser.wait(function () {
+                return browser.isElementPresent(faqPage.userMenuDropdown);
+            }, faqPage.mainElementsWaitTime);
+
+            faqPage.administrationOptions.first().click();
+        }).then(function () {
+            // wait for modal iframe to appear
+            browser.wait(function () {
+                return browser.isElementPresent(faqPage.sideMenuIframe);
+            }, faqPage.iframeWaitTime);
+
+            // switch to sidebar menu iframe
+            browser.switchTo().frame(browser.findElement(
+                By.css('.cms_sideframe-frame iframe')));
+
+            browser.wait(function () {
+                return browser.isElementPresent(faqPage.pagesLink);
+            }, faqPage.mainElementsWaitTime);
+
+            faqPage.pagesLink.click();
+
+            // check if the page already exists and return the status
+            return faqPage.addPageLink.isPresent();
+        }).then(function (present) {
+            if (present === true) {
+                // page is absent - create new page
+                browser.wait(function () {
+                    return browser.isElementPresent(faqPage.addPageLink);
+                }, faqPage.mainElementsWaitTime);
+
+                faqPage.addPageLink.click();
+
+                browser.wait(function () {
+                    return browser.isElementPresent(faqPage.titleInput);
+                }, faqPage.mainElementsWaitTime);
+
+                faqPage.titleInput.sendKeys('Test').then(function () {
```

Comment:
Tried several ways - still the same result - error breaks the test execution. So, one nesting level can be removed, others will stay as they are